infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/infrastructure/msgHandlers/LogReplicationAbstractServer.java

```diff
@@ -0,0 +1,97 @@
+package org.corfudb.infrastructure.logreplication.infrastructure.msgHandlers;
+
+import lombok.extern.slf4j.Slf4j;
+import org.corfudb.infrastructure.AbstractServer;
+import org.corfudb.infrastructure.logreplication.transport.IClientServerRouter;
+import org.corfudb.protocols.service.CorfuProtocolMessage;
+import org.corfudb.runtime.proto.service.CorfuMessage;
+
+import java.util.concurrent.atomic.AtomicReference;
+
+import static org.corfudb.protocols.CorfuProtocolServerErrors.getNotReadyErrorMsg;
+import static org.corfudb.protocols.service.CorfuProtocolMessage.getHeaderMsg;
+import static org.corfudb.protocols.service.CorfuProtocolMessage.getResponseMsg;
+
+@Slf4j
+public abstract class LogReplicationAbstractServer {
+
+    /**
+     * Current server state
```

Comment:
IMO we don't need this abstract class, since we are not extending it with multiple servers like Corfu side

Reply:
I agree, This and the IClientServerRouter both need not be abstracted out. They don't have multiple implementations.

I wanted to run by the team about this